packages/@lwc/engine-dom/src/custom-elements/create-custom-element.ts

```diff
@@ -53,34 +53,33 @@ function createBaseUpgradableConstructor() {
         }
 
         connectedCallback() {
+            // native `connectedCallback`/`disconnectedCallback` are only enabled in native lifecycle mode
             if (isTrue(nativeLifecycleElementsToUpgradedByLWC.get(this))) {
                 connectRootElement(this);
             }
         }
+
         disconnectedCallback() {
+            // native `connectedCallback`/`disconnectedCallback` are only enabled in native lifecycle mode
             if (isTrue(nativeLifecycleElementsToUpgradedByLWC.get(this))) {
                 disconnectRootElement(this);
             }
         }
+
         formAssociatedCallback(form: HTMLFormElement | null) {
-            if (isTrue(nativeLifecycleElementsToUpgradedByLWC.get(this))) {
-                runFormAssociatedCallback(this, form);
-            }
+            runFormAssociatedCallback(this, form);
         }
+
         formDisabledCallback(disabled: boolean) {
-            if (isTrue(nativeLifecycleElementsToUpgradedByLWC.get(this))) {
-                runFormDisabledCallback(this, disabled);
-            }
+            runFormDisabledCallback(this, disabled);
         }
+
         formResetCallback() {
-            if (isTrue(nativeLifecycleElementsToUpgradedByLWC.get(this))) {
-                runFormResetCallback(this);
-            }
+            runFormResetCallback(this);
         }
+
         formStateRestoreCallback(state: FormRestoreState | null, reason: FormRestoreReason) {
-            if (isTrue(nativeLifecycleElementsToUpgradedByLWC.get(this))) {
-                runFormStateRestoreCallback(this, state, reason);
-            }
+            runFormStateRestoreCallback(this, state, reason);
```

Comment:
This is the only actual function change in this PR. Everything else is just tests.